doc/dev/tests.md

```diff
@@ -0,0 +1,378 @@
+        assert schema_properties.version is 1
+        assert schema_properties.serialization_type == "Avro"
+
+        with pytest.raises(HttpResponseError):
+            client.get_schema('a' * 32)
+```
+The `AzureTestCase` class has the ability to define a client by passing in the client object and the account URL, without having to worry about identity. Test files should not import `azure.identity`, the `self.create_basic_client` will take care of loading environment variables and creating the default credentials.
+
+The test infrastructure heavily leverages the `assert` keyword, which tests if the condition following it is true, and if it is not the program will raise an `AssertionError`. When writing tests, any uncaught exception results in a failure, from an assert or from the code itself (ie. `TypeError`, `ValueError`, `HttpResponseError`, etc.). The assert statements are testing that all the exected properties of the returned object are not `None`, and the last two assert statements verify that the tested properties are a given value. The last two lines of the test use a [context manager](https://docs.python.org/3/library/contextlib.html) used from the `pytest` library that tests whether the following block of code will raise a certain exception.
+
+## Run and record the test
+
+From your terminal run the `pytest` command to run all the tests that you have written so far.
+
+```cmd
+(env)azure-sdk-for-python\sdk\api-learn\implementation-tutorial> pytest
+```
+
+Your update should run smooth and have green dots representing passing tests. Now if you look at the contents of your `tests` directory there should be a new directory called `recording` with four `.yaml` files. Each `yaml` file is a recording for a single test. To run a test in playback mode change the `testsettings_local.cfg` to `live-mode: false` and rerun the tests with the same command. The test infrastructure will use the `.yaml` recordings to mock the HTTP traffic and run the tests.
+
+
+## More Test Examples
+
+This section will demonstrate how to write tests with the `devtools_testutils` package with a few samples to showcase the features of the test framework.
+
+### Example 1: Basic Recording and Interactions
+
+```python
+import os
+import pytest
+
+from azure.data.tables import TableServiceClient
+from devtools_testutils import AzureTestCase
+
+class ExampleStorageTestCase(AzureTestCase):
+
+    def test_create_table(self):
+        credential = self.get_credential(TableServiceClient)
+        client = self.create_client_from_credential(TableServiceClient, credential, account_url=self.account_url)
+
+        with pytest.raises(ValueError):
+            table_name = "an_invalid_table"
+            created = client.create_table(table_name)
+```
+
+This simple tests that the client verifies a valid table name before sending the HTTP request (Azure Data Tables can only be alphanumeric values). This test will have no recording associated with it.
+
+Notes:
```

Comment:
All these notes seems unnecessary?

Reply:
Yes those are leftover notes. thanks for the catch